addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -618,276 +606,692 @@ private void onDisconnected() {
+                        logger.error(
+                                "{}: context='{}' Bridge OFFLINE because ZoneMinder Server OPT_TRIGGER was disabled",
+                                getLogIdentifier(), context);
+                        zmConnectStatus = ZoneMinderConnectionStatus.SERVER_OPT_TRIGGERS_DISABLED;
+                        newStatus = ThingStatus.OFFLINE;
+                        statusDetail = ThingStatusDetail.CONFIGURATION_ERROR;
+                        statusDescription = "Option external triggers 'OPT_TRIGGERS' was disabled";
+                        updateBridgeStatus(newStatus, statusDetail, statusDescription, false);
+                        return;
+                    }
+                    // Check if ZoneMinder Server API can be accessed
+                    else if (!serverProxy.isApiEnabled()) {
+                        logger.error("{}: context='{}' Bridge OFFLINE because ZoneMinder Server API was disabled",
+                                getLogIdentifier(), context);
+                        zmConnectStatus = ZoneMinderConnectionStatus.SERVER_API_DISABLED;
+                        newStatus = ThingStatus.OFFLINE;
+                        statusDetail = ThingStatusDetail.CONFIGURATION_ERROR;
+                        statusDescription = "ZoneMinder Server API was disabled";
+                        updateBridgeStatus(newStatus, statusDetail, statusDescription, false);
+                        return;
+                    }
+
+                } finally {
+                    releaseSession();
                 }
-                // Verify that 'OPT_TRIGGER' is set to true in ZoneMinder
-                else if (!serverProxy.isTriggerOptionEnabled()) {
-                    newStatus = ThingStatus.OFFLINE;
-                    statusDetail = ThingStatusDetail.CONFIGURATION_ERROR;
-                    statusDescription = "ZoneMinder Server option 'OPT_TRIGGERS' not enabled";
-                    updateBridgeStatus(newStatus, statusDetail, statusDescription);
-                    return;
-                } else {
-                    // Seems like everything is as we want it :-)
-                    _isOnline = true;
+
+                newStatus = ThingStatus.ONLINE;
+                statusDetail = ThingStatusDetail.NONE;
+                statusDescription = "";
+                updateBridgeStatus(newStatus, statusDetail, statusDescription, true);
+
+                // Ask all child things to update their Availability Status
+                for (Thing thing : getThing().getThings()) {
+                    ZoneMinderBaseThingHandler thingHandler = (ZoneMinderBaseThingHandler) thing.getHandler();
+                    if (thingHandler instanceof ZoneMinderThingMonitorHandler) {
+                        try {
+                            thingHandler.updateAvaliabilityStatus(getZoneMinderConnection());
+                        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.